aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/UpdateHandler.java

```diff
@@ -22,88 +32,145 @@
+    ) {
+        final ResourceModel desiredModel = progress.getResourceModel();
+
+        if (desiredModel.getPolicyDocument() != null) {
+            return ProgressEvent.progress(desiredModel, callbackContext);
+        }
+
+        return proxy.initiate("AWS-CodeArtifact-Repository::Update::DeleteRepositoryPermissionsPolicy", proxyClient,
+            progress.getResourceModel(), progress.getCallbackContext())
+            .translateToServiceRequest(Translator::translateDeletePermissionsPolicyRequest)
+            .makeServiceCall((awsRequest, client) -> {
+                DeleteRepositoryPermissionsPolicyResponse awsResponse = null;
+                try {
+                    awsResponse = client.injectCredentialsAndInvokeV2(awsRequest, client.client()::deleteRepositoryPermissionsPolicy);
+                    logger.log("Repository permission policy successfully deleted.");
+                } catch (final AwsServiceException e) {
+                    String domainName = desiredModel.getDomainName();
+                    Translator.throwCfnException(e, Constants.PUT_REPOSITORY_POLICY, domainName);
+                }
+                return awsResponse;
+            })
+            .stabilize((awsRequest, awsResponse, client, model, context) -> policyIsDeleted(model, client))
+            .progress();
+    }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> disassociateExternalConnections(
+        ProgressEvent<ResourceModel, CallbackContext> progress,
+        final CallbackContext callbackContext,
+        final ResourceHandlerRequest<ResourceModel> request,
+        ProxyClient<CodeartifactClient> proxyClient,
+        Set<String> externalConnectionsToRemove,
+        Logger logger
+    ) {
+        ResourceModel model = request.getDesiredResourceState();
+        logger.log(String.format("Removing external connections: %s", externalConnectionsToRemove.toString()));
+
+        if (CollectionUtils.isNullOrEmpty(externalConnectionsToRemove)) {
+            logger.log("No external connections to remove.");
+            // Nothing to remove, continue
+            return ProgressEvent.progress(model, callbackContext);
+        }
+
+        // Loop is currently not necessary because only 1 external connection is allowed, leaving this in for
+        // when multiple are supported
+        externalConnectionsToRemove.forEach(ec -> {
+            try {
+                DisassociateExternalConnectionRequest disassociateExternalConnectionRequest = Translator.translateDisassociateExternalConnectionsRequest(model, ec);
+                proxyClient.injectCredentialsAndInvokeV2(disassociateExternalConnectionRequest, proxyClient.client()::disassociateExternalConnection);
+            } catch (final AwsServiceException e) {
+                String repositoryName = progress.getResourceModel().getRepositoryName();
+                Translator.throwCfnException(e, Constants.ASSOCIATE_EXTERNAL_CONNECTION, repositoryName);
```

Comment:
are these constants supposed to be based onthe put operations (same as the above comment) or was this just a copy pasta issue :)

Reply:
copy pasta issue!